### ISSUE TITLE
qt 5.11 breaks fermentables

### ISSUE THREAD

**Issue by mikfire** [Brewtarget/brewtarget#416]
qt 5.11 seems to have broken the fermentable table for postgresql. Everything else display correctly, but the FermentableTable is empty, as is the fermentable tree. 

I've examined the database, and it all seems good. This is either something at the display layer, or there is an issue with how we are querying teh database. The only thing I see on the console is:
```QPSQLDriver::getResult: Query results lost - probably discarded on executing another SQL query.```

Google doesn't provide much insight, but does suggest it has to do with the q.setForwardOnly method. I've tried commenting those lines out, but there was no change.

**Comment by mikfire** [comment]
Ah. Yeah. All the template methods in database.h. 

If you comment out all of the ```q.setForwardOnly(true)``` lines in that file then I can see all of my fermentables. 

Digging a bit harder, and I see this in the Qt documentation:
```
Warning: PostgreSQL: While navigating the query results in forward-only mode, 
do not execute any other SQL command on the same database connection. 
This will cause the query results to be lost.
```

That would probably be a good indication of what is going on, and I think we have gone to great lengths to make sure we have only one database connection. I am not up to troubleshooting concurrency issues tonight.

**Comment by mikfire** [comment]
Ok. This seems to be a side effect of  #353 . I am curious about that commit. 

One of the original design ideas @rocketman768  and I had when we moved to the database model was no cached data -- all reads and writes are done without caching in memory. It seems the referenced commit changed that, but for Fermentables only.

The problem is that we enable forwardOnly() when doing the initial load via Database::populateElements. Calling new Fermentable causes about 20 more queries to be executed, which causes the Postgresql driver to invalidate the results in populateElements and we end up loading just one fermentable. My guess is that this is broken in many version of Qt.

So why did we break the pattern, and break it only for fermentables?

**Comment by mikfire** [comment]
This was all fixed as part of #453
